### PR TITLE
Pandaria Isolation

### PR DESCRIPTION
**Pull request by zumbak04**
## Developer changelog:
- Pandaria is isolated now. Pandaria characters can't declare wars, send councilors, build trade posts, marry someone outside of Pandaria. The Mists of Pandaria are parted around the 613th year. A game rule to regulate the Mists is included.

## Script changelog (ignore if you can't read CK2 script):
- Replaced `is_within_diplo_range` where both right and left scopes are characters by is_within_prev_diplo_range_trigger that checks if characters are in Pandaria.

# How to test:
Enter the game, test the game rule and try to wait for the 613th year.

### CODE REVIEW

File: localisation/00_zuwc_united_localisation.csv [hunk 1]
```diff
@@ -3041,16 +3041,22 @@ nick_the_unstoppable;the Unstoppable;;;;;;;;;;;;;x
...
-;;;;;;;;;;;;;;x
-;;;;;;;;;;;;;;x
-;;;;;;;;;;;;;;x
-;;;;;;;;;;;;;;x
-;;;;;;;;;;;;;;x
-;;;;;;;;;;;;;;x
-;;;;;;;;;;;;;;x
+mists_of_pandaria_dont_block_interactions_tooltip;Mists of Pandaria do NOT block interactions;;;;;;;;;;;;;x
+EVTTITLE_WCPAN_100;The Mists of Pandaria;;;;;;;;;;;;;x
+EVTDESC_WCPAN_100_azeroth;This morning the messenger brought you astounding news. A group of boats fishing in the South Seas has stumbled upon a land where it shouldn't be. One of the ships wrecked, but the rest managed to find a way out of mists.\n\nAll ships were aware of avoiding those waters due to dense fog and sea legends, but it looks like the mists have parted.\n\nThe fishermen even tell that they have seen gold towers far on the horizon. If their stories aren't fake, they could discover a new land inhabited by sentient beings right south of the Maelstrom. We should send expeditions to make sure.;;;;;;;;;;;;;x
```
Comment (LordSilvermane) [inline localisation/00_zuwc_united_localisation.csv:3046]:
```suggestion
EVTDESC_WCPAN_100_azeroth;This morning the messenger brought you astounding news. A group of boats fishing in the South Seas has stumbled upon an uncharted land. One of the ships was wrecked, but the rest managed to find a way out of mists.\n\nAll ships were aware of the importance of avoiding those waters due to dense fog and old fables, but it looks like the mists have parted.\n\nThe fishermen even tell that they have seen gold towers far on the horizon. If their stories are true, they could discover a new land inhabited by sentient beings directly south of the Maelstrom. We should send expeditions to learn more.;;;;;;;;;;;;;x
```

File: localisation/00_zuwc_united_localisation.csv [hunk 2]
```diff
@@ -3041,16 +3041,22 @@ nick_the_unstoppable;the Unstoppable;;;;;;;;;;;;;x
...
+EVTOPTA_WCTRO_100_conquer;New lands to explore and conquer.;;;;;;;;;;;;;x
+EVTOPTA_WCTRO_100_welcome;Welcome, wanderers!;;;;;;;;;;;;;x
+EVTOPTA_WCPAN_100_expeditions;Send expeditions, I want to know more.;;;;;;;;;;;;;x
```
Comment (LordSilvermane) [inline localisation/00_zuwc_united_localisation.csv:3049]:
```suggestion
EVTOPTA_WCPAN_100_expeditions;Send an expedition, I want to know more.;;;;;;;;;;;;;x
```

File: localisation/00_zuwc_united_localisation.csv [hunk 3]
```diff
@@ -3041,16 +3041,22 @@ nick_the_unstoppable;the Unstoppable;;;;;;;;;;;;;x
...
+EVTDESC_WCPAN_100_pandaria;This morning the messenger brought you astounding news. A group of lighthouse keepers has noticed strange boats coming out of the fog. One of the ships wrecked, but the rest managed to find a way out of the mists.\n\nHow could these boats find a way to Pandaria and out? May it mean that the mists have parted?\n\nThe wrecked ship's crew is not like any other race we know. They speak about kingdoms and places beyond the sea. If their stories aren't fake, the world is much bigger and more dangerous than we thought. We must be prepared to face more ships and wanderers soon.;;;;;;;;;;;;;x
```
Comment (LordSilvermane) [inline localisation/00_zuwc_united_localisation.csv:3050]:
```suggestion
EVTDESC_WCPAN_100_pandaria;This morning the messenger brought you astounding news. A number of lighthouse keepers have noticed strange boats coming out of the fog. One of the ships was wrecked, but the rest managed to find a way out of the mists.\n\nHow could these boats have found a way in and out of Pandaria's mists? Could it mean that the mists have parted?\n\nThe wrecked ship's crew is not like any other race we know. They speak about kingdoms and lands beyond the sea. If their stories are true, the world is much bigger and more dangerous than we believed. We must be prepared to face more ships and wanderers soon.;;;;;;;;;;;;;x
```

File: localisation/00_zuwc_united_localisation.csv [hunk 4]
```diff
@@ -3041,16 +3041,22 @@ nick_the_unstoppable;the Unstoppable;;;;;;;;;;;;;x
...
+EVTDESC_WCCLN_113;Unfortunately, the marriage between [target_other_spouse.GetDynName] and [target_spouse.GetDynName] is impossible. [Root.GetDivorceReason];;;;;;;;;;;;;x
+GetDivorceReason_mists;You sent messengers to [target_other_spouse.GetFirstName], but they disappeared in the mists of the South Seas.;;;;;;;;;;;;;x
+GetDivorceReason_cant_marry;Due to [target_spouse.GetHerHis] race, [target_spouse.GetFirstName] can't marry and love someone [target_spouse.GetSheHe] wish.;;;;;;;;;;;;;x
+GetDivorceReason_wrong_race;Due to the different physiology and incompatibility of their races, they can't marry and love each other.;;;;;;;;;;;;;x
+RULE_MISTS_OF_PANDARIA;�RWarCraft�!: Mists of Pandaria;;;;;;;;;;;;;x
+RULE_MISTS_OF_PANDARIA_ON;On;;;;;;;;;;;;;x
+RULE_MISTS_OF_PANDARIA_ON_DESC;The Mists of Pandaria are enabled, making Pandaria isolated till they're gone around the 613th year.;;;;;;;;;;;;;x
```
Comment (LordSilvermane) [inline localisation/00_zuwc_united_localisation.csv:3057]:
```suggestion
RULE_MISTS_OF_PANDARIA_ON_DESC;The Mists of Pandaria are enabled, making Pandaria isolated until they part around the 613th year.;;;;;;;;;;;;;x
```

File: localisation/00_zuwc_united_localisation.csv [hunk 5]
```diff
@@ -3041,16 +3041,22 @@ nick_the_unstoppable;the Unstoppable;;;;;;;;;;;;;x
...
+RULE_MISTS_OF_PANDARIA_OFF;Off;;;;;;;;;;;;;x
+RULE_MISTS_OF_PANDARIA_OFF_DESC;The Mists of Pandaria are disabled thus Pandaria isn't isolated.;;;;;;;;;;;;;x
```
Comment (LordSilvermane) [inline localisation/00_zuwc_united_localisation.csv:3059]:
```suggestion
RULE_MISTS_OF_PANDARIA_OFF_DESC;The Mists of Pandaria are disabled, making Pandaria open to the world.;;;;;;;;;;;;;x
```

Comment (zumbak04) [reply localisation/00_zuwc_united_localisation.csv]: Done in 33a5b5203922313bb6a9824fef205b46499ecbcb.

**Comment by zumbak04** [comment]
> Loc looks good. I could not marry, build trade posts, or declare wars outside of pandaria. However, I could send my councilors on missions outside the mists. Everything works as intended post the parting of the mists.

Fixed in ee4b0e961c75a7031144c1e1279f590dd71dbb2b.

**Pull request merged**